src/platform/react-hooks/src/hooks/useChannel.ts

```diff
@@ -45,13 +45,16 @@ export function useChannel(
   const channelOptionsRef = useRef(channelOptions);
   const ablyMessageCallbackRef = useRef(ablyMessageCallback);
 
-  const channel = useMemo(() => ably.channels.get(channelName, channelOptionsRef.current), [ably, channelName]);
+  const channel = useMemo(
+    () => ably.channels.get(channelName, channelOptionsWithAgent(channelOptionsRef.current)),
```

Comment:
I am a little bit concerned that if we always provide options parameter and you somehow use same channel twice it will throw an exception. And it's not react way, because it prevents hooks composition (with composition it is easy to render the same hook twice, without even realizing it). Can we do smth like this? 

```js
if (ably.channel.exists(channelName) {
  const channel = ably.channel.get(channelName);
  channel.setOptions(channelOptionsWithAgent(channelOptionsRef.current));
  return channel;
} else {
  return ably.channels.get(channelName, channelOptionsWithAgent(channelOptionsRef.current)),
}
```

or maybe we can introduce additional option, similar for [react-use-websocket's share](https://www.npmjs.com/package/react-use-websocket#share-boolean)?

Reply:
yeah i was thinking about this and really it seems like a limitation of ably-js, the code snippet you shared would stop an error but it would still interfere with other components using the same channel so doesn't seem ideal :/

As for the `share` option, this is kinda what we already do - if you call `channels.get` with a channel that's already being used then it will use that channel instead of creating a new instance.

Reply:
>As for the share option, this is kinda what we already do - if you call channels.get with a channel that's already being used then it will use that channel instead of creating a new instance.

yeah, but with current changes, `options` can't be empty it will always contain `agent` param, and will throw an exception, won't it?

Reply:
it will only throw an exception if you call `useChannel` with different options in two or more different components, which was always how it behaved - i don't think this changes the error cases